sdks/python/apache_beam/pipeline.py

```diff
@@ -1353,18 +1353,28 @@ def transform_to_runner_api(
       environment_id = context.get_environment_id_for_resource_hints(
           self.resource_hints)
 
+    inputs = {
```

Comment:
> p | beam.WindowInto(...) should also be an error right there (checked in WindowInto's expand method).

+1 to Robert's comment, moving this validation inside `WindowInto` methods  would be better.